src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1457,6 +1464,13 @@ private bool FlushTrackerQueue(IContextTracker contextTracker)
                     WriteObject(match.Line);
                 }
             }
+            else if (OnlyMatching)
+            {
+                foreach (MatchInfo match in contextTracker.EmitQueue)
+                {
+                    WriteObject(match.Matches[0].Value);
```

Comment:
Do we only want to return the value of the _first_ match, or should we be returning the value from all the matches in the line?

Reply:
There's also the `-AllMatch` switch so it could do both depending on if that's specified or not as we'll only have 1 match per line without `-AllMatch` anyway.

Reply:
Yeah, that would probably be ideal here :)

Reply:
Done.